internal/services/orbital/contact_resource.go

```diff
@@ -0,0 +1,205 @@
+	return map[string]*schema.Schema{}
+}
+
+func (r ContactResource) ModelObject() interface{} {
+	return &ContactResourceModel{}
+}
+
+func (r ContactResource) ResourceType() string {
+	return "azurerm_orbital_contact"
+}
+
+func (r ContactResource) Create() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 30 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			var model ContactResourceModel
+			if err := metadata.Decode(&model); err != nil {
+				return err
+			}
+
+			client := metadata.Client.Orbital.ContactClient
+			subscriptionId := metadata.Client.Account.SubscriptionId
+
+			spacecraftId, err := contact.ParseSpacecraftID(model.Spacecraft)
+			if err != nil {
+				return err
+			}
+			id := contact.NewContactID(subscriptionId, model.ResourceGroup, spacecraftId.SpacecraftName, model.Name)
+			existing, err := client.Get(ctx, id)
+			if err != nil && !response.WasNotFound(existing.HttpResponse) {
+				return fmt.Errorf("checking for presence of existing %s: %+v", id, err)
+			}
+
+			if !response.WasNotFound(existing.HttpResponse) {
+				return metadata.ResourceRequiresImport(r.ResourceType(), id)
+			}
+
+			contactProfile := contact.ResourceReference{
+				Id: &model.ContactProfileId,
+			}
+
+			contactProperties := contact.ContactsProperties{
+				ContactProfile:       contactProfile,
+				GroundStationName:    model.GroundStationName,
+				ReservationEndTime:   model.ReservationEndTime,
+				ReservationStartTime: model.ReservationStartTime,
+			}
+
+			contact := contact.Contact{
+				Id:         utils.String(id.ID()),
+				Name:       utils.String(model.Name),
+				Properties: &contactProperties,
+			}
+			if _, err = client.Create(ctx, id, contact); err != nil {
+				return fmt.Errorf("creating %s: %+v", id, err)
+			}
+			metadata.SetID(id)
+			return nil
+		},
+	}
+}
+
+func (r ContactResource) Read() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 5 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			client := metadata.Client.Orbital.ContactClient
+			id, err := contact.ParseContactID(metadata.ResourceData.Id())
+			if err != nil {
+				return err
+			}
+
+			resp, err := client.Get(ctx, *id)
+			if err != nil {
+				if response.WasNotFound(resp.HttpResponse) {
+					return metadata.MarkAsGone(id)
+				}
+				return fmt.Errorf("reading %s: %+v", *id, err)
+			}
+
+			spacecraftId := contact.NewSpacecraftID(id.SubscriptionId, id.ResourceGroupName, id.SpacecraftName)
+			if model := resp.Model; model != nil {
+				props := model.Properties
+				state := ContactResourceModel{
+					Name:                 id.ContactName,
+					ResourceGroup:        id.ResourceGroupName,
+					Spacecraft:           spacecraftId.ID(),
+					ReservationStartTime: props.ReservationStartTime,
+					ReservationEndTime:   props.ReservationEndTime,
+					GroundStationName:    props.GroundStationName,
+					ContactProfileId:     *props.ContactProfile.Id,
```

Comment:
this'll be a crash so we'll need to nil-check this

Reply:
Added nil check.